### PR TITLE
Implement Younger Monocot Pipeline

### PR DESCRIPTION
**Pull request by eberrigan**
- Added `YoungerMonocotPipeline` as subclass of `Pipeline`
- Test `YoungerMonocotPipeline`

### Summary by CodeRabbit

- New Feature: Introduced `YoungerMonocotPipeline` class for trait computation of younger monocot plants, including root lengths, angles, intersections, and network length.
- Refactor: Updated `get_grav_index` function in `sleap_roots/lengths.py` to handle both scalar and array inputs uniformly, improving code flexibility and robustness.
- Test: Added comprehensive test cases for `get_grav_index` function covering various scenarios such as scalar inputs, array inputs, shape mismatch, NaN values, and zero lengths.
- Test: Implemented a new test function `test_younger_monocot_pipeline` to validate the functionality of the newly introduced `YoungerMonocotPipeline` class.

### CODE REVIEW

File: tests/test_lengths.py [hunk 1]
```diff
@@ -146,7 +146,7 @@ def lengths_all_nan():
     return np.array([np.nan, np.nan, np.nan])
 
 
-# test get_grav_index function
+# tests for get_grav_index function
 def test_get_grav_index(canola_h5):
     series = Series.load(
         canola_h5, primary_name="primary_multi_day", lateral_name="lateral_3_nodes"
```
[the remark this answers is not in the document]
Comment (eberrigan) [reply tests/test_lengths.py:152]: It looks complete to me.

File: sleap_roots/trait_pipelines.py [hunk 2]
```diff
@@ -899,3 +899,475 @@ def get_initial_frame_traits(self, plant: Series, frame_idx: int) -> Dict[str, A
...
+            TraitDef(
+                name="main_grav_indices",
+                fn=get_base_tip_dist,
+                input_traits=["main_base_pts", "main_tip_pts"],
+                scalar=False,
+                include_in_csv=True,
+                kwargs={},
+                description="Straight-line distance(s) from the base(s) to the"
+                "tip(s) of the main root(s).",
+            ),
```
[the remark this answers is not in the document]
Comment (eberrigan) [reply sleap_roots/trait_pipelines.py:1200]: Done

File: tests/test_trait_pipelines.py [hunk 3]
```diff
@@ -18,3 +18,64 @@ def test_dicot_pipeline(canola_h5, soy_h5):
...
+        all_traits["main_grav_indices_mean_median"] >= 0
+    ).all(), "main_grav_indices_mean_median in all_traits contains negative values"
+    assert (
+        (0 <= rice_traits["main_angles_proximal_p95"])
+        & (rice_traits["main_angles_proximal_p95"] <= 180)
+    ).all(), "angle_column in rice_traits contains values out of range [0, 180]"
+    assert (
+        (0 <= all_traits["main_angles_proximal_median_p95"])
+        & (all_traits["main_angles_proximal_median_p95"] <= 180)
+    ).all(), "angle_column in all_traits contains values out of range [0, 180]"
```
[the remark this answers is not in the document]
Comment (eberrigan) [reply tests/test_trait_pipelines.py]: Done

**Review by talmo — APPROVED** [review]
Looks great to me!

**Pull request merged**